pkg/stargate/config.go

```diff
@@ -58,6 +59,26 @@ func CreateStargateConfigMap(namespace, cassandraYaml, stargateCqlYaml string, d
 	}
 }
 
+func CreateVectorConfigMap(namespace, vectorToml string, dc cassdcapi.CassandraDatacenter) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      stargateapi.VectorAgentConfigMapNameStargate(dc.Spec.ClusterName, dc.Name),
+			Namespace: namespace,
+			Labels: map[string]string{
+				k8ssandra.NameLabel:                      k8ssandra.NameLabelValue,
+				k8ssandra.PartOfLabel:                    k8ssandra.PartOfLabelValue,
+				k8ssandra.ComponentLabel:                 k8ssandra.ComponentLabelValueStargate,
+				k8ssandra.CreatedByLabel:                 k8ssandra.CreatedByLabelValueK8ssandraClusterController,
+				k8ssandra.K8ssandraClusterNameLabel:      dc.Labels[k8ssandra.K8ssandraClusterNameLabel],
```

Comment:
@adutra I have update the code so that the Stargate Vector ConfigMap's controller reference is set to the Stargate object here:
https://github.com/k8ssandra/k8ssandra-operator/pull/819/files#diff-2e5c443ca862a5be41c3c80f344d80d6e0b83c0084c8724a0364330699d5df6fR55

which is the same thing we do for the Stargate ConfigMap here:
https://github.com/k8ssandra/k8ssandra-operator/blob/5040b3fb075c09312f57bfbfd49c9b0c86f3b63b/controllers/stargate/stargate_controller.go#L437